seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/scheduler/PipelineBaseScheduler.java

```diff
@@ -56,18 +68,22 @@ public void startScheduling() {
                     handlePipelineStateUpdateError(pipeline, PipelineState.SCHEDULED);
                     return null;
                 }
-                if (!applyResourceForPipeline(pipeline)) {
-                    return null;
+                Map<PhysicalVertex, SlotProfile> slotProfiles;
+                try {
+                    slotProfiles = applyResourceForPipeline(pipeline);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
                 }
+                pipeline.whenComplete((state, error) -> releasePipelineResource(Lists.newArrayList(slotProfiles.values())));
```

Comment:
I don't think we should release resources when the pipeline complete. In design, the state of pipeline is controlled by `taskFuture` in `PhysicalVertex`, so I think we should release resources in `taskFuture.whenComplete()`

Reply:
If like this, when retry task, should request resource again, this is not acceptable